File: 017)AnimeBook/app/src/main/java/com/example/animebook/AnimeAdapter.java
package com.example.animebook;

import android.content.Context;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.databinding.DataBindingUtil;
import androidx.recyclerview.widget.RecyclerView;

import com.example.animebook.databinding.RecyclerRowBinding;

import java.util.List;

public class AnimeAdapter extends RecyclerView.Adapter<AnimeAdapter.AnimeHolder>{
    private Context mContext;
    private List<Anime> animeList;

    public AnimeAdapter(Context mContext, List<Anime> animeList) {
        this.mContext = mContext;
        this.animeList = animeList;
    }

    @NonNull
    @Override
    public AnimeHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater layoutInflater = LayoutInflater.from(mContext);
        RecyclerRowBinding binding = DataBindingUtil.inflate(layoutInflater, R.layout.recycler_row, parent, false);
        return new AnimeHolder(binding);
    }

    @Override
    public void onBindViewHolder(@NonNull AnimeHolder holder, int position) {
        Anime anime = animeList.get(position);
        RecyclerRowBinding binding = holder.binding;
        binding.textViewRV.setText(anime.getName());
        binding.cardViewLine.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(mContext, AnimeActivity.class);
                intent.putExtra("info", "old");
                intent.putExtra("animeId", anime.getId());
                mContext.startActivity(intent);
            }
        });
    }

    @Override
    public int getItemCount() {
        if (animeList == null){
            return 0;
        }else {
            return animeList.size();
        }
    }

    public class AnimeHolder extends RecyclerView.ViewHolder{
        private RecyclerRowBinding binding;

        public AnimeHolder(RecyclerRowBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }
    }
}

File: 057)KotlinAnimeBook/app/src/main/java/com/elyesasimsek/kotlinanimebook/AnimeAdapter.kt
package com.elyesasimsek.kotlinanimebook

import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.RecyclerView
import com.elyesasimsek.kotlinanimebook.databinding.RecyclerRowBinding

class AnimeAdapter (
    private var context: Context,
    private var animeList: List<Anime>
): RecyclerView.Adapter<AnimeAdapter.AnimeHolder>(){
    class AnimeHolder(val binding: RecyclerRowBinding): RecyclerView.ViewHolder(binding.root)

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): AnimeHolder {
        val layoutInflater = LayoutInflater.from(context)
        val binding = DataBindingUtil.inflate<RecyclerRowBinding>(layoutInflater, R.layout.recycler_row, parent, false)
        return AnimeHolder(binding)
    }

    override fun getItemCount(): Int {
        if (animeList == null){
            return 0;
        }else{
            return animeList.size
        }
    }

    override fun onBindViewHolder(holder: AnimeHolder, position: Int) {
        val anime = animeList[position]
        val binding = holder.binding

        binding.textViewRV.text = anime.name
        binding.cardViewLine.setOnClickListener {
            val intent = Intent(context, AnimeActivity::class.java)
            intent.putExtra("info", "old")
            intent.putExtra("id", anime.id)
            context.startActivity(intent)
        }
    }

}